src/sagas/helpers.js

```diff
@@ -344,3 +344,24 @@ export function* progressiveRetryRequest(request, maxRetries = MAX_RETRIES) {
   // throw last error after retries exhausted
   throw lastError;
 }
+
+/**
+ * A wrapper effect to catch unexpected error and provide a binding
+ * for the desired handling behavior.
+ *
+ * @param {Object} effect The targeted effect.
+ * @param {(error) => void} onError The error handling effect,
+ *    which receives the error object as first argument.
+ *
+ * @returns An anonymous effect.
+ */
+export function safeEffect(effect, onError) {
+  // eslint-disable-next-line func-names
+  return function* (payload) {
```

Comment:
question(non-blocking): Why not name this function? Wouldn't it help in an eventual debugging session?

Reply:
I'm giving it a name `_safeEffect`: 554868188bf4f0fc3418d82d7eb12c1153c50298. Thank you!